andela/might-guy-rc
# #144361487 Display real-time results as shopper is typing

#### What does this PR do?
- It implements the feature "Display real-time results as shopper is typing"

#### Description of Task to be completed?
- Similar to a google search, display relevant results on the page as a user is typing rather than wait until they submit the search.

#### How should this be manually tested?
- Login and type what you want to search for in the text area provided which appears after you click on the search icon. The results would be displayed as seen in the screenshots below.

#### Any background context you want to provide?
- N/A

#### What are the relevant pivotal tracker stories?
- #144361487 Display real-time results as shopper is typing

#### Screenshots (if appropriate)
### Products
<img width="1280" alt="screen shot 2017-05-17 at 12 37 06 pm" src="https://cloud.githubusercontent.com/assets/26273354/26155889/94a357e0-3b0c-11e7-9db6-d02211d5590c.png">

### Search examples
<img width="1280" alt="screen shot 2017-05-17 at 1 42 50 pm" src="https://cloud.githubusercontent.com/assets/26273354/26155939/b7aa77aa-3b0c-11e7-83cd-6af05e93e4e0.png">
<img width="1252" alt="screen shot 2017-05-17 at 1 43 03 pm" src="https://cloud.githubusercontent.com/assets/26273354/26155992/cca65228-3b0c-11e7-8d06-6fc102e0eb15.png">
<img width="1268" alt="screen shot 2017-05-17 at 1 43 32 pm" src="https://cloud.githubusercontent.com/assets/26273354/26155998/d0fb61ba-3b0c-11e7-8fde-24a57f4f6b99.png">

### Search result when no product with title is found
<img width="1252" alt="screen shot 2017-05-17 at 1 44 11 pm" src="https://cloud.githubusercontent.com/assets/26273354/26156049/f4d06144-3b0c-11e7-8781-d8a3dfc0f95d.png">
<img width="1272" alt="screen shot 2017-05-17 at 1 45 02 pm" src="https://cloud.githubusercontent.com/assets/26273354/26156052/f7a35818-3b0c-11e7-8f64-f5196c34e989.png">



#### Questions:
- N/A
